### ISSUE TITLE
Service File Issues

### ISSUE THREAD

**Issue by n8rade** [jonathanspw/headscale-rpm#5]
I found that to get Headscale to work properly on AlmaLinux 9.3 I had to change one of the service file's `After` from:
~~~
[Unit]
After=syslog.target
After=network.target
~~~
To:
~~~
[Unit]
After=syslog.target
After=network-online.target
~~~

I was running into issues where my Headscale node wouldn't act as an exit node/subnet router properly (when also configured as a tailscale client to itself). After the change it works flawlessly for me. I was able to narrow it down to this by performing a service restart of the headscale service specifically after a reboot to then get things to work properly again.

I can't fully explain why this works for me, only that it does. I've tested this with 3 reboots so far and it continues to work flawlessly as I undo some of the other troubleshooting I was doing with the config.

Feel free to close if this might be a one off issue for whatever reason, but I did want to at least report it as I spent a good amount of time troubleshooting the config.yaml thinking it was a headscale configuration issue.

**Comment by jonathanspw** [comment]
Yes you're 100% right, this needs to be network-online.target.  I'll fix this soon.

**Comment by jonathanspw** [comment]
Fixed in 0.22.3-3 via https://github.com/jonathanspw/headscale-rpm/commit/a4a7fe2a6c2e8d230f9fe1d60fe042153ecd87d8
